hera_mc/rtp.py

```diff
@@ -168,6 +170,115 @@ def create(cls, time, obsid, event):
         return cls(time=time, obsid=obsid, event=event)
 
 
+class RTPTaskProcessEvent(MCDeclarativeBase):
+    """
+    Definition of rtp_task_process_event table.
+
+    Attributes
+    ----------
+    time : BigInteger Column
+        GPS time of this status, floored. The primary key.
+    obsid : BigInteger Column
+        Observation obsid.  Part of primary_key. Foreign key into
+        Observation table.
+    task_name : String Column
+        Name of task in pipeline (e.g., OMNICAL). Part of primary key.
+    event : Enum Column
+        One of ["queued", "started", "finished", "error"] (rtp_process_enum).
+
+    """
+
+    __tablename__ = 'rtp_task_process_event'
+    time = Column(BigInteger, primary_key=True)
+    obsid = Column(BigInteger, ForeignKey('hera_obs.obsid'), primary_key=True)
+    task_name = Column(Text, primary_key=True)
+    event = Column(
+        Enum(*rtp_task_process_enum, name="rtp_task_process_enum"),
+        nullable=False,
+    )
+
+    @classmethod
+    def create(cls, time, obsid, task_name, event):
+        """
+        Create a new rtp_task_process_event object.
+
+        Parameters
+        ----------
+        time : astropy Time object
+            Time of this event
+        obsid : long
+            Observation obsid (Foreign key into Observation)
+        task_name : str
+            Name of task in pipeline (e.g., OMNICAL).
+        event : {"queued", "started", "finished", "error"}
+            Process event type.
+
+        Returns
+        -------
+        RTPTaskProcessEvent object
+
+        """
+        if not isinstance(time, Time):
+            raise ValueError('time must be an astropy Time object')
+        time = floor(time.gps)
```

Comment:
should we enforce the enum here? Downside is that if something else comes in it will error and not get recorded. Maybe we don't want to throw errors...

Reply:
I lean toward not enforcing it, because right now everything that uses it _should_ be resecting the enum already. My guess is right now we'd get an SQLAlchemy-related error right now, which is hopefully informative enough.

Reply:
That's fine, just thought it was worth thinking about.